pages/[category]/index.tsx

```diff
@@ -19,30 +19,40 @@ const CategoryPage = () => {
   return (
     <section className="flex min-h-[calc(100%-68px)] flex-col">
       <div className=" m-auto items-start gap-2">
-        <div className="m-auto gap-2 flex flex-col items-center justify-center ">
-          <p className="md:text-4xl text-xl uppercase font-bold text-theme-secondary dark:text-gray-text w-fit mx-auto text-center">
-            <span className="text-theme-primary">-/</span>
-            {category}
-          </p>
-          <p className="text-xl hidden md:block">
-            Get access to all exclusive{' '}
-            <span className="capitalize text-theme-primary">
-              {category}
-            </span>{' '}
-            resources!
-          </p>
-        </div>
-        <div className="flex flex-wrap gap-3 mt-6 justify-center items-start mx-auto lg:max-w-sm max-md:max-w-sm">
-          {subcategories.map((subcat, i) => (
-            <Link
-              key={i}
-              href={`/${category}${subcat.url}`}
-              className={`border-2 dark:border-theme-primary rounded-md bg-transparent dark:text-theme-primary text-text-secondary px-3 py-1 text-md capitalize hover:border-dashed hover:text-light-primary shadow-lg`}
-            >
-              {subcat.name}
-            </Link>
-          ))}
-        </div>
+        {subcategories?.length ? (
+          <>
+            <div className="m-auto gap-2 flex flex-col items-center justify-center ">
+              <p className="md:text-4xl text-xl uppercase font-bold text-theme-secondary dark:text-gray-text w-fit mx-auto text-center">
+                <span className="text-theme-primary">-/</span>
+                {category}
+              </p>
+              <p className="text-xl hidden md:block">
+                Get access to all exclusive{' '}
+                <span className="capitalize text-theme-primary">
+                  {category}
+                </span>{' '}
+                resources!
+              </p>
+            </div>
+            <div className="flex flex-wrap gap-3 mt-6 justify-center items-start mx-auto lg:max-w-sm max-md:max-w-sm">
+              {subcategories.map((subcat, i) => (
+                <Link
+                  key={i}
+                  href={`/${category}${subcat.url}`}
+                  className={`border-2 dark:border-theme-primary rounded-md bg-transparent dark:text-theme-primary text-text-secondary px-3 py-1 text-md capitalize hover:border-dashed hover:text-light-primary shadow-lg`}
+                >
+                  {subcat.name}
+                </Link>
+              ))}
+            </div>
+          </>
+        ) : (
+          <div className='text-center'>
+            <p className="text-theme-secondary dark:text-gray-text w-fit mx-auto text-center">
+              <span className='md:text-7xl text-xl uppercase font-bold '>404|</span> No resources found.
+            </p>
+          </div>
```

Comment:
The request's response status itself should be 404. Here the request is ok (200), but we are rendering a 404 message. The correct way to go about it is to implement `getStaticPaths` with `fallback: false`. That way request to a route that is not included in `getStaticPaths` will 404 and by default nextjs will render the message. 

You can check the `[...subcategory].tsx` file for reference.

Reply:
i guess its okay..

Reply:
It's a bug.